code/controllers/subsystem/garbage.dm

```diff
@@ -0,0 +1,436 @@
+SUBSYSTEM_DEF(garbage)
+	name = "Garbage"
+	priority = FIRE_PRIORITY_GARBAGE
+	wait = 2 SECONDS
+	flags = SS_POST_FIRE_TIMING|SS_BACKGROUND|SS_NO_INIT
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+	init_order = INIT_ORDER_GARBAGE
+
+	var/list/collection_timeout = list(0, 2 MINUTES, 10 SECONDS)	// deciseconds to wait before moving something up in the queue to the next level
+
+	//Stat tracking
+	var/delslasttick = 0			// number of del()'s we've done this tick
+	var/gcedlasttick = 0			// number of things that gc'ed last tick
+	var/totaldels = 0
+	var/totalgcs = 0
+
+	var/highest_del_time = 0
+	var/highest_del_tickusage = 0
+
+	var/list/pass_counts
+	var/list/fail_counts
+
+	var/list/items = list()			// Holds our qdel_item statistics datums
+
+	//Queue
+	var/list/queues
+
+
+/datum/controller/subsystem/garbage/PreInit()
+	queues = new(GC_QUEUE_COUNT)
+	pass_counts = new(GC_QUEUE_COUNT)
+	fail_counts = new(GC_QUEUE_COUNT)
+	for(var/i in 1 to GC_QUEUE_COUNT)
+		queues[i] = list()
+		pass_counts[i] = 0
+		fail_counts[i] = 0
+
+/datum/controller/subsystem/garbage/stat_entry(msg)
+	var/list/counts = list()
+	for(var/list/L in queues)
+		counts += length(L)
+	msg += "Q:[counts.Join(",")]|D:[delslasttick]|G:[gcedlasttick]|"
+	msg += "GR:"
+	if(!(delslasttick+gcedlasttick))
+		msg += "n/a|"
+	else
+		msg += "[round((gcedlasttick/(delslasttick+gcedlasttick))*100, 0.01)]%|"
+
+	msg += "TD:[totaldels]|TG:[totalgcs]|"
+	if(!(totaldels+totalgcs))
+		msg += "n/a|"
+	else
+		msg += "TGR:[round((totalgcs/(totaldels+totalgcs))*100, 0.01)]%"
```

Comment:
Spacing in operators.

As a sidenote, I wonder if we ought to ignore /tg/ and make the debug info more readable over compact- because it gets frankly kinda hard to read with everything crammed together and acronymed like this.